### ISSUE TITLE
Firmware Debugging Improvements

### ISSUE THREAD

**Issue by tracernz** [d-ronin/dRonin#1387]
Make the process a little smoother, more flexible, easier. Possibly use https://github.com/mbedmicro/pyOCD instead of OpenOCD. OpenOCD is a bit of a problem to install by script on Windows (no official distributions, unofficial ones require 7-zip).

**Comment by mlyle** [comment]
Relates to #209?

**Comment by tracernz** [comment]
Ah, yep. The PR associated with #209 (from Tau) already merged IIRC. Windows is still in a bad state though (and is problematic to fix, hence this issue).

**Comment by mlyle** [comment]
Is it so unreasonable to say "if you want to debug firmware on windows, figure it out yourself?"

**Comment by tracernz** [comment]
No, it's fine, but if there's a good solution that isn't too much work/is potentially better on other platforms too, I'm willing to spend some time on it when I'm bored. :grin:
